x-pack/plugins/lens/public/react_embeddable/renderer/lens_embeddable_component.tsx

```diff
@@ -57,9 +57,10 @@ export function LensEmbeddableComponent({
   const rootRef = useDispatcher(hasRendered, api);
 
   // Publish the data attributes only if avaialble/visible
-  const title = api.hidePanelTitle?.getValue()
-    ? undefined
-    : { 'data-title': api.panelTitle?.getValue() ?? api.defaultPanelTitle?.getValue() };
+  const title =
+    api.hidePanelTitle?.getValue() || internalApi.getDisplayOptions()?.noPanelTitle
+      ? undefined
+      : { 'data-title': api.panelTitle?.getValue() ?? api.defaultPanelTitle?.getValue() };
```

Comment:
I think in this case `||` is the best choice over of the nullish coaleshing operator to avoid introducing a bug:
* if `hidePanelTitle` is `undefined` then it's equivalent to `false`, so `||` behaves like `??`
* if `hidePanelTitle` is `false` then the `??` would stop there and not even evaluate the computed `displayOptions` 🐛 
* if `hidePanelTitle` is `true` then it would behave like the previous case

It may make sense to either invert the order AND use the nullish coalesce operator, or make `getDisplayOptions()` method encapsulate the entire logic for title hiding internally and not leak anything here.

I like the latter, wdyt?

Reply:
you are right, definitely better the second suggestion, thanks